File: Java_Virtual_World/src/Gameplay/Animals/Turtle.java
package Gameplay.Animals;

import Gameplay.Organism;
import java.lang.Math;

public class Turtle extends Animal{
    public static final int STRENGTH = 2;
    public static final int INITIATIVE = 1;
    public static final double PROBABILITY = 0.25;

    public Turtle(int x, int y){
        super(x ,y , STRENGTH, INITIATIVE);
    }

    @Override
    public void Action(){
        if(Math.random() < PROBABILITY){
            randomMove(1);
        }
    }
    @Override
    public boolean didBlocked(Organism opponent){
        return opponent.getStrength() < strength;
    }
    @Override
    public String Draw(){
        return "🐢";
    }
    @Override
    public Organism Copy(){
        return new Turtle(x,y);
    }
    @Override
    public String toString(){
        return "Turtle";
    }
}

File: Python_Virtual_World/Turtle.py
from random import random

from MyVector import MyVector
from Animal import Animal
from Organism import Organism


class Turtle(Animal):


    STRENGTH = 2
    INITIATIVE = 1
    PROPABILITY = 0.25
    DEFENSE = 5


    def __init__(self, position : MyVector):
        super().__init__(position, Turtle.STRENGTH, Turtle.INITIATIVE)


    def action(self):

        if random() < Turtle.PROPABILITY:
            self._RandomMove()


    def didBlocked(self,other : Organism) -> bool:

        if other.getStrength() < Turtle.DEFENSE:
            self._world.getNots().add("Turtle has blocked the " + other.__str__())
            return True
        return False



    def draw(self) -> str:

        return "#438C7E"


    def __str__(self):
        return "Turtle"
